Classes/Private/Manager/Video/OCTVideoEngine.h

```diff
@@ -0,0 +1,100 @@
+//
+//  OCTVideoEngine.h
+//  objcTox
+//
+//  Created by Chuong Vu on 6/19/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "OCTToxAV.h"
+
+@interface OCTVideoEngine : NSObject
+
+@property (weak, nonatomic) OCTToxAV *toxav;
+
+/**
+ * Current friend number that video engine should
+ * process video data to and from.
+ */
+@property (nonatomic, assign) OCTToxFriendNumber friendNumber;
+
+/**
+ * This must be called prior to using the video session.
+ * @param error Pointer to error object.
+ * @return YES if successful, otherwise NO.
+ */
+- (BOOL)setupWithError:(NSError **)error;
+
+/**
+ * Process incoming video frames.
+ * Set to YES to process video, otherwise NO
+ * to ignore the incoming frames in receiveVideoFrameWithWidth...
+ * Will be set to YES automatically when calling startSendingVideo
+ * Will be set to NO after stopSendingVideo
+ */
+@property (nonatomic, assign) BOOL processIncomingVideo;
+
+/**
+ * Start sending video data.
+ * This will turn on processIncomingVideo to YES
+ */
+- (void)startSendingVideo;
+
+/**
+ * Stop sending video data.
+ * This will turn off processIncomingVideo to NO
+ */
+- (void)stopSendingVideo;
+
+/**
+ * Indicates if the video session is running.
+ * @return YES if running, NO otherwise.
+ */
+- (BOOL)isVideoSessionRunning;
+
+/**
+ * Current video feed.
+ */
+- (UIView *)videoFeed;
```

Comment:
view.layer? :-)

    Dmytro

> On 31 Jul 2015, at 18:47, chuong vu <notifications@github.com> wrote:
> 
> In Classes/Private/Manager/Video/OCTVideoEngine.h:
> 
> > +/**
> > + * Stop sending video data.
> > + * This will turn off processIncomingVideo to NO
> > + */
> > +- (void)stopSendingVideo;
> > +
> > +/**
> > + * Indicates if the video session is running.
> > + * @return YES if running, NO otherwise.
> > + */
> > +- (BOOL)isVideoSessionRunning;
> > +
> > +/**
> > + * Current video feed.
> > + */
> > +- (UIView *)videoFeed;
> Hmm I'll have to look into that. Not sure if there's a way to grab CALayer from GLKView instead.
> 
> —
> Reply to this email directly or view it on GitHub.
>